File: apps/ios/HaruBook/Features/Parents/WeeklyReportViewModel.swift
import Foundation
import Observation

@Observable
@MainActor
final class WeeklyReportViewModel {
    private(set) var reports: [ParentalProfileReport] = []
    var isLoading: Bool = false
    var error: String?

    func load() async {
        isLoading = true
        defer { isLoading = false }
        do {
            let response: ParentalReportResponse = try await APIClient.shared.send(
                Endpoint(path: "/api/parents/report", method: .get),
            )
            self.reports = response.report
            self.error = nil
        } catch {
            self.error = "리포트를 불러오지 못했어요. 잠시 후 다시 시도해 주세요."
        }
    }
}

File: apps/android/app/src/main/java/site/smap/harubook/features/parents/WeeklyReportViewModel.kt
package site.smap.harubook.features.parents

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.flow.update
import kotlinx.coroutines.launch
import site.smap.harubook.core.models.ParentalProfileReport
import site.smap.harubook.core.models.ParentalReportResponse
import site.smap.harubook.core.networking.ApiClient

data class WeeklyReportUiState(
    val reports: List<ParentalProfileReport> = emptyList(),
    val isLoading: Boolean = false,
    val error: String? = null,
)

class WeeklyReportViewModel : ViewModel() {
    private val _state = MutableStateFlow(WeeklyReportUiState())
    val state: StateFlow<WeeklyReportUiState> = _state.asStateFlow()

    fun load() {
        viewModelScope.launch {
            _state.update { it.copy(isLoading = true, error = null) }
            try {
                val response: ParentalReportResponse = ApiClient.get("/api/parents/report")
                _state.update { it.copy(reports = response.report, isLoading = false) }
            } catch (e: Throwable) {
                _state.update {
                    it.copy(isLoading = false, error = "리포트를 불러오지 못했어요. 잠시 후 다시 시도해 주세요.")
                }
            }
        }
    }
}
